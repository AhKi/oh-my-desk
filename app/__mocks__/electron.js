export default jest.fn();

export const webContents = {
  getAllWebContents: jest.fn(() => []),
};

export const ipcMain = {
  on: jest.fn(),
};

export const BrowserWindow = jest.fn(info => ({
  close: jest.fn(),
  loadURL: jest.fn(),
  once: jest.fn(),
  on: jest.fn(),
  setAlwaysOnTop: jest.fn(),
  show: jest.fn(),
  showInactive: jest.fn(),
  getContentBounds: jest.fn(() => ({
    x: info.x,
    y: info.y,
    height: info.height,
    width: info.width,
  })),
  webContents: {
    on: jest.fn(),
    send: jest.fn(),
    openDevTools: jest.fn(),
  },
}));

export const ipcRenderer = {
  on: jest.fn(),
  send: jest.fn(),
};

export const remote = {
  getGlobal: jest.fn(),
  getCurrentWindow: jest.fn(),
  Menu: {
    buildFromTemplate: jest.fn(() => ({
      popup: jest.fn(),
    })),
  },
  app: {
    getVersion: jest.fn(),
  },
};

export const shell = {
  openExternal: jest.fn(),
};

export const app = {
  getName: jest.fn(() => 'oh-my-desk'),
  getPath: jest.fn(arg => arg),
  getVersion: jest.fn(),
  getAppPath: jest.fn(() => 'mock-path'),
  isReady: jest.fn(),
  on: jest.fn(),
};

export const Menu = {
  buildFromTemplate: jest.fn(() => ({
    popup: jest.fn(),
  })),
  setApplicationMenu: jest.fn(),
};

export const globalShortcut = {
  register: jest.fn(),
  unregister: jest.fn(),
};
