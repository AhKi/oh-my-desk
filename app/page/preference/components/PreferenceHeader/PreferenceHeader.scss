.PreferenceHeader {
  background-color: rgb(246,246,246);
  text-align: center;
  height: 55px;
  width: 100%;
}

.PreferenceHeader__Link {
  float: left;
  font-size: 13px;
  height: 55px;
  padding: 5px;
}

.PreferenceHeader__Link--select {
  background-color: rgb(232, 232, 232);
}
