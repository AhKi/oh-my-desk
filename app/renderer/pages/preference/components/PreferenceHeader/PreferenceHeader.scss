.PreferenceHeader {
  background-color: rgb(227,227,227);
  text-align: center;
  height: 80px;
  width: 100%;
}

.PreferenceHeader__Draggable {
  background-color: rgb(227,227,227);
  height: 21px;
  -webkit-app-region: drag;
}

.PreferenceHeader__nav {
  display: flex;
}

.PreferenceHeader__Link {
  font-size: 13px;
  height: 55px;
  padding: 5px;
  margin: 2px;
  border-radius: 15%;
}

.PreferenceHeader__Link--select {
  background-color: rgb(195, 195, 195);
}

.PreferenceHeader__icon {
  width: 45px;
  height: 45px;
}
