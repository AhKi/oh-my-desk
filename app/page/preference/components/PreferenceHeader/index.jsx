import React from 'react';
import { NavLink } from 'react-router-dom';
import i18n from 'constants/i18n';
// import PropTypes from 'prop-types';
import './PreferenceHeader.scss';

const propTypes = {};
const defaultProps = {};

class PreferenceHeader extends React.Component {
  render() {
    const text = i18n().preference;
    return (
      <div className="PreferenceHeader">
        {process.platform === 'darwin' && <div className="PreferenceHeader__Draggable" />}
        <NavLink
          activeClassName="PreferenceHeader__Link--select"
          className="PreferenceHeader__Link"
          exact
          to="/"
        >
          {text.main}
        </NavLink>
        <NavLink
          activeClassName="PreferenceHeader__Link--select"
          className="PreferenceHeader__Link"
          to="/update"
        >
          {text.update}
        </NavLink>
        <NavLink
          activeClassName="PreferenceHeader__Link--select"
          className="PreferenceHeader__Link"
          to="/hot-key"
        >
          {text.hotKey}
        </NavLink>
        <NavLink
          activeClassName="PreferenceHeader__Link--select"
          className="PreferenceHeader__Link"
          to="/widget"
        >
          {text.widget}
        </NavLink>
      </div>
    );
  }
}

PreferenceHeader.propTypes = propTypes;
PreferenceHeader.defaultProps = defaultProps;

export default PreferenceHeader;
