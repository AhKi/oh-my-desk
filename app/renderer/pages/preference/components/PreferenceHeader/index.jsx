import React from 'react';
import os from 'os';
import { NavLink } from 'react-router-dom';
import i18n from 'constants/i18n';
import './PreferenceHeader.scss';

const propTypes = {};
const defaultProps = {};

class PreferenceHeader extends React.Component {
  render() {
    const text = i18n().preference;
    return (
      <div className="PreferenceHeader">
        {os.platform() === 'darwin' && <div className="PreferenceHeader__Draggable" />}
        <NavLink
          activeClassName="PreferenceHeader__Link--select"
          className="PreferenceHeader__Link"
          exact
          to="/"
        >
          {text.main}
        </NavLink>
      </div>
    );
  }
}

PreferenceHeader.propTypes = propTypes;
PreferenceHeader.defaultProps = defaultProps;

export default PreferenceHeader;
