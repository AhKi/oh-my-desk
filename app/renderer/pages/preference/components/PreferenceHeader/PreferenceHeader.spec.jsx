import React from 'react';
import { shallow } from 'enzyme';
import { remote } from 'electron';
import os from 'os';

import PreferenceHeader from '.';

jest.mock('os');
os.platform = () => 'darwin';

describe('<PreferenceHeader />', () => {
  remote.getGlobal = jest.fn(() => () => JSON.stringify({ config: { language: 'English' } }));

  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<PreferenceHeader />);

    expect(wrapper).toMatchSnapshot();
  });
});
