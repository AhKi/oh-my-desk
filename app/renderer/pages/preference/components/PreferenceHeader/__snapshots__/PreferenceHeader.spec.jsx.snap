// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreferenceHeader /> should match to snapshot when render default 1`] = `
<div
  className="PreferenceHeader"
>
  <div
    className="PreferenceHeader__nav"
  >
    <NavLink
      activeClassName="PreferenceHeader__Link--select"
      aria-current="page"
      className="PreferenceHeader__Link"
      exact={true}
      to="/"
    >
      <SVGInline
        className="PreferenceHeader__icon"
        classSuffix="-svg"
        cleanup={Array []}
        cleanupExceptions={Array []}
        component="span"
        svg=""
      />
    </NavLink>
  </div>
</div>
`;
